frontend/src/metabase/admin/tasks/components/Logs/hooks.ts

```diff
@@ -46,15 +46,17 @@ export function usePollingLogsQuery(pollingDurationMs: number) {
 
   // keep track of mounted state to avoid settings state after unmount
   // clear timeout that is polling for logs
-  useMount(() => {
-    isMountedRef.current = true;
-    fetchLogs();
-    pollingInterval.start();
-    return () => {
+  useLifecycles(
```

Comment:
I'm kind of on the fence about using more hooks from `react-use` than necessary.

On one hand, this is the only usage of `useLifecycles` in the entire project. And it's convenient because we don't need to deal with dependencies if we use `useEffect`.

On the other hand, I think it would be more familiar to more React developers to use `useEffect` with `[]` in the dependencies to achieve the same effect.

I'll refer this opinion to other @metabase/frontend-developers

Reply:
I noticed this too, my vote goes to a useEffect with `[]` as deps, I think it is more immediate to read

Reply:
I'm on the other side here - we can't really use `useEffect` with `[]` as deps since we have `react/exhaustive-deps` - but I think we should use `useMount` and `useUnmount` explicitly, like we do in quite a few times in other parts of the code. I like the idea of `useLifecycles` but I think it's a little much in terms of combining things

Reply:
we need to ignore that eslint rule, and it'd work. However, I agree that we're using `useMount` and `useUnmount` in a bunch of places, so it could be justified to do that as well.

To sum up, I think there are 2 paths forward.
1. Use `useEffect` with `[]` with eslint-ignore comment OR
2. use `useMount` and `useUnmount` separately

I don't have any strong conviction on any approach but from this point, the second one starts to look a bit more appealing since we don't have to explicitly disable that ESLint rule.

Reply:
Please don't ignore the lint rule and use a ref or a proper hook

Reply:
I'm not in the metabase team but I'd like to also give my opinion 😅 

I agree with @ranquild because the intent from (1) is not clear, a dev might look at `eslint-ignore` and inadvertently remove it because it's not a best practice to ignore those rules, especially this one. So I'd choose the second path.

There is a third alternative using `useEffect` with correct and memoized dependencies.

A bit off topic here, while I was thinking about the memoized dependencies, I think that there's another bug [here](https://github.com/metabase/metabase/blob/811f6bd26a7b00f7caa3d9644adf60fe741af776/frontend/src/metabase/admin/tasks/components/Logs/hooks.ts#L45)
After each re-render we start a new `timer` and the old one is not stopped because we didn't unmount the component. I'll try to reproduce it and open a new issue separately.

Reply:
I vote for using `useMount` and `useUnmount`